tests/components/adguard/test_service.py

```diff
@@ -0,0 +1,184 @@
+                        "filters": [
+                            {
+                                "name": "Example",
+                                "url": "https://example.com/1.txt",
+                            }
+                        ],
+                    },
+                },
+                {
+                    "method": "post",
+                    "url": "https://127.0.0.1:3000/control/filtering/set_url",
+                    "json": {
+                        "whitelist": False,
+                        "url": "https://example.com/1.txt",
+                    },
+                },
+            ],
+            {"url": "https://example.com/1.txt"},
+        ),
+        (
+            SERVICE_ENABLE_URL,
+            [
+                {
+                    "method": "get",
+                    "url": "https://127.0.0.1:3000/control/filtering/status",
+                    "json": {
+                        "filters": [
+                            {
+                                "name": "Example",
+                                "url": "https://example.com/1.txt",
+                            }
+                        ],
+                    },
+                },
+                {
+                    "method": "post",
+                    "url": "https://127.0.0.1:3000/control/filtering/set_url",
+                    "json": {
+                        "whitelist": False,
+                        "url": "https://example.com/1.txt",
+                    },
+                },
+            ],
+            {"url": "https://example.com/1.txt"},
+        ),
+        (
+            SERVICE_REFRESH,
+            [
+                {
+                    "method": "post",
+                    "url": "https://127.0.0.1:3000/control/filtering/refresh?force=false",
+                    "json": {"whitelist": False},
+                }
+            ],
+            {"force": False},
+        ),
+        (
+            SERVICE_REMOVE_URL,
+            [
+                {
+                    "method": "post",
+                    "url": "https://127.0.0.1:3000/control/filtering/remove_url",
+                    "json": {
+                        "whitelist": False,
+                        "url": "https://example.com/1.txt",
+                    },
+                }
+            ],
+            {"url": "https://example.com/1.txt"},
+        ),
+    ],
+)
+async def test_service(
+    hass: HomeAssistant,
+    aioclient_mock: AiohttpClientMocker,
+    mock_config_entry: MockConfigEntry,
+    service: str,
+    mocked_requests: list[dict],
+    service_call_data: dict,
+) -> None:
+    """Test the adguard services be unregistered with unloading last entry."""
+    for mocked_request in mocked_requests:
+        aioclient_mock.request(
+            mocked_request["method"],
+            mocked_request["url"],
+            json=mocked_request["json"],
+            headers={"Content-Type": CONTENT_TYPE_JSON},
+        )
```

Comment:
Can we make sure we patch out the library instead of mocking http calls?

Reply:
rework done :heavy_check_mark: